lib/tessel/wifi.js

```diff
@@ -98,8 +98,43 @@ Tessel.prototype.connectToNetwork = function(opts) {
       })
       .then(function() {
         // Then set the new credientials
-        logs.info('Credentials set!');
-        resolve();
+        return self.connection.exec(commands.commitWirelessCredentials());
+      })
+      .then(function(remoteProcess) {
+        // Once the credentials have been comitted
+        remoteProcess.once('close', function() {
+          // Restart the wifi
+          return self.connection.exec(commands.reconnectWifi())
+            .then(function(remoteProcess) {
+              // Once the wifi restart process closes
+              return self.receive(remoteProcess).then(function() {
+                logs.info('Credentials set. Checking connection...');
+                self.connection.exec(commands.ubusListen())
+                  .then(function(remoteProcess) {
+                    remoteProcess.stdout.on('data', function(data) {
+                      if (data.toString().indexOf('ifup') > -1) {
+                        logs.info('Successfully connected!');
+                        // End the connection
+                        return self.connection.end()
+                          .then(resolve);
+                      }
+                    });
+                    remoteProcess.stderr.on('data', function(data) {
+                      logs.info('Error connecting:', data.toString());
+                      // End the connection
+                      return self.connection.end()
+                        .then(resolve);
+                    });
```

Comment:
This change is missing: https://github.com/tessel/t2-cli/pull/445#discussion_r44868229